department-of-veterans-affairs/vets-website
# provider and date reset

**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?

## Summary

This PR resets both the provider and selected slot in redux and the slot stored in session when someone visits the schedule-referral page. To that end it also updates our mock generators to produce multiple providers and it also swaps the session storage to be slotId based rather than date based. Plus it brings that slot selection into our area of the reducer.

I also made a few changes to get the back button working so that I could manually test this. In the process, I also refactored a few functions into utils to help with readability. 

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#98701

## Testing done

Manual testing and unit tests.

## What areas of the site does it impact?

Community Care referral scheduling in VAOS

## Acceptance criteria
 - [ ] Provider redux data is retained when moving forward past the date selector page and doesn't refetch when navigating back to page
 - [ ] Slot selection s retained when moving forward past the date selector page and is saved when navigation back to page
 - [ ] When navigating back to the schedule referral page and then forward again, provider is re-fetched and saved slot is reset
 
### Quality Assurance & Testing


### Error Handling


### Code Review steps
 - turn on `vaOnlineSchedulingCCDirectScheduling` feature locally
 - run mock and app locally
 - navigate to `http://localhost:3001/my-health/appointments/schedule-referral/date-time?id=add2f0f4-a1ea-4dea-a504-a54ab57c6800`
 - select a date/time and then click `continue`
 - click the back link in the bread crumb of the next page
 - observe that the selection is still made and there there was no API call
 - click the back link in the bread crumb again
 - click the `Schedule your appointment` action link
 - observe that the selection is no longer saved and there was an API call to get fresh provider data
